### ISSUE TITLE
format suggests lint-only --disregard-sqlfluffignores option

### ISSUE THREAD

**Issue by acdha** [sqlfluff/sqlfluff#6818]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I was testing the latest version on a file which had been ignored while waiting for #901 and got this error message which lead to a confusing situation:

```bash
$ sqlfluff format --disregard-sqlfluffignores ¬.sql
==== finding fixable violations ====
WARNING    Exact file path ….sql was given but it was ignored by an ignore pattern set in .sqlfluffignore, re-run with `--disregard-sqlfluffignores` to not process ignore files. 
==== no fixable linting violations found ====
All Finished 📜 🎉!
$ sqlfluff format --disregard-sqlfluffignores ¬.sql
Usage: sqlfluff format [OPTIONS] [PATHS]...
Try 'sqlfluff format -h' for help.

Error: No such option: --disregard-sqlfluffignores
```

The cause is simple: `--disregard-sqlfluffignores` is only declared on the `lint` subcommand: https://github.com/sqlfluff/sqlfluff/blob/3cf5754aaa95ad65786cc59d15d8eedf329496e3/src/sqlfluff/cli/commands.py#L576-L580

### Expected Behaviour

`format` would work like `lint`

### Observed Behaviour

```
Error: No such option: --disregard-sqlfluffignores
```

### How to reproduce

Run with a file which is listed in the `.sqlfluffignore` file.

### Dialect

MySQL

### Version

sqlfluff, version 3.4.0

### Configuration

n/a

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by prakharb10** [comment]
Ran into the same scenario and issue today
